bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,839 @@
+                    if (result != null) {
+                        final Response resp = result.getResponse();
+                        if (resp != null) {
+                            scheduler.execute(() -> {
+                                handleNon200Response(resp);
+                            });
+                        }
+                    }
+                }
+            }
+        };
+
+        sendAsyncSiteApiRequest(httpClient, daily, uplinkBridge.getApiKey(), location, listener);
+    }
+
+    protected static void sendAsyncSiteApiRequest(final HttpClient httpClient, final boolean daily, final String apiKey,
+            final PointType location, final Response.CompleteListener listener) {
+        final String url = ((daily) ? GET_FORECAST_URL_DAILY : GET_FORECAST_URL_HOURLY)
+                .replace("<LATITUDE>", location.getLongitude().toString())
+                .replace("<LONGITUDE>", location.getLongitude().toString());
+
+        final Request request = httpClient.newRequest(url).method(HttpMethod.GET)
+                .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_TYPE.toString())
+                .header(GET_FORECAST_API_KEY_HEADER, apiKey).timeout(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        request.send(listener);
+    }
+
+    private static String calculatePrefix(final String prefix, final int plusOffset) {
+        final StringBuilder strBldr = new StringBuilder(26);
+        strBldr.append(prefix);
+        if (plusOffset > 0) {
+            strBldr.append(GROUP_POSTFIX_BOTH_FORECASTS);
+            if (plusOffset < 10) {
+                strBldr.append("0");
+            }
+            strBldr.append(plusOffset);
+        }
+        strBldr.append(GROUP_PREFIX_TO_ITEM);
+        return strBldr.toString();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        config = getConfigAs(MetOfficeDataHubSiteApiConfiguration.class);
+
+        if (config.location.isBlank()) {
+            @Nullable
+            PointType userLocation = locationProvider.getLocation();
+            if (userLocation == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        getLocalizedText("site.error.no-user-location"));
+                return;
+            } else {
+                location = userLocation;
+            }
+        } else {
+            try {
+                location = new PointType(config.location);
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        getLocalizedText("site.error.invalid-location"));
+                return;
+            }
+        }
+
+        /**
+         * Setup the initial device's status
+         */
+        initTask = scheduler.schedule(() -> {
+            updateStatus(ThingStatus.ONLINE);
+            checkDataRequired();
+            reconfigurePolling();
```

Comment:
This can be removed as `checkDataRequired` checks if any channels are linked and cofnigures the polling accordingly
```suggestion
```

Reply:
Still checking but I picked up I believe a few extra unnecessary call's hopefully this is included.